src/opening-period/page/CreateNewOpeningPeriodPage.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Button, Notification } from 'hds-react';
+import { useHistory } from 'react-router-dom';
+import formatDate from 'date-fns/format';
+import parse from 'date-fns/parse';
+import api, { Resource } from '../../common/utils/api/api';
+import { ResourceInfo } from '../../resource/page/ResourcePage';
+import Datepicker from '../../components/datepicker/Datepicker';
+import './CreateNewOpeningPeriodPage.scss';
+import { ResourceState } from '../../common/lib/types';
+
+type Inputs = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string;
+  openingPeriodEndDate: string;
+};
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+export default function CreateNewOpeningPeriodPage({
+  resourceId,
+}: {
+  resourceId: string;
+}): JSX.Element {
+  const { register, handleSubmit, errors } = useForm<Inputs>({
+    mode: 'all',
+    defaultValues: {
+      openingPeriodTitle: '',
+      openingPeriodOptionalDescription: '',
+      openingPeriodBeginDate: undefined,
+      openingPeriodEndDate: undefined,
+    },
+  });
+  const history = useHistory();
+  const [resource, setResource] = useState<Resource>();
+  const [hasLoadingResourceError, setLoadingResourceError] = useState<
+    Error | undefined
+  >(undefined);
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [periodBeginDate, setPeriodBeginDate] = useState<Date | null>(null);
+  const [periodEndDate, setPeriodEndDate] = useState<Date | null>(null);
+  const [submitStatus, setSubmitStatus] = useState<SubmitStatus>('init');
+
+  const onSubmit = async (data: Inputs): Promise<void> => {
+    try {
+      if (!resource) {
```

Comment:
This one probably requires a better solution :sweat_smile:.

Problem here is:

If I remove the `if (!resource)` clause, then eslint will complain about row 52 that resource could be null/undefined:

![image](https://user-images.githubusercontent.com/2777633/99534077-873efd80-29af-11eb-9d95-fc186b1692fe.png)

I could try to solve it with inserting postfix '?' like this: `resource: resource?.id`. But then eslint will complain:

![image](https://user-images.githubusercontent.com/2777633/99534400-07656300-29b0-11eb-929d-8bc073f38a9d.png)

I think the problem is that `api.postDatePeriod` requires object that complies with DatePeriod object. And DatePeriod object always has value for resource. We could fix it so that we make it optional in the DatePeriod type definition. But Im not sure if that would give the correct signal. Because DatePeriod as an object got from database always has a resource related to it. It's like something it always should have. But since we open this page we do not yet necessarily know the resource id, because we load the resource info during the page load. So in the code it actually gets a value undefined in row 37 (default value is undefined). Maybe there's some workaround for this rather than throwing a new error?

We could set the resource field of DatePeriod type definition to optional, but I feel it would give wrong information about the object, since in practice it should always have a resource field value other than undefined.
Maybe there's some easy typescript way to handle this that I just missed because of typescript-noobness :sweat_smile: